### ISSUE TITLE
when CDPATH environment variable is set running /bin/start-micro-quickstart fails 

### ISSUE THREAD

**Issue by vimil-saju** [apache/druid#12878]
### Affected Version

druid-0.23.0

### Description

when `CDPATH` environment variable is set example scripts like `/bin/start-micro-quickstart` do not run because the below lines in `run-druid` and `run-zk` scripts

```
CONFDIR="$(cd "$CONFDIR" && pwd)"
WHEREAMI="$(cd "$WHEREAMI" && pwd)"
```
expects `cd` command to return no output. however when `CDPATH` environment variable is set  the `cd` command returns the current directory path as output. This causes the `CONFDIR `and `WHEREAMI` variables to have the directory path twice ( the output from cd command gets concatenated to the output from pwd command)

below is a quick test to validate the issue.

```
CDPATH=".:~"
T=$(cd Temp && pwd)
echo $T
```

the output will be `~/Temp ~/Temp` instead of `~/Temp`
assuming there is Temp dir under the home directory

**Comment by FrankChen021** [comment]
> however on macos the cd command returns the current directory path as output.

I don't know this before, but for each released version, some of developers test the release candidates on various platforms including macOS, but have never reported such problem.

**Comment by FrankChen021** [comment]
BTW BigSur was used on my macbookpro half a year ago, and I have never encountered such problem. Now I'm using macOS Monterey.

**Comment by vimil-saju** [comment]
> BTW BigSur was used on my macbookpro half a year ago, and I have never encountered such problem. Now I'm using macOS Monterey.

Digging a bit deeper I found this was the case if the environment variable CDPATH was set. below is a quick test you can run.

```
CDPATH=".:~"
T=$(cd Temp && pwd)
echo $T
 ```
 the output will be `~/Temp ~/Temp` instead of `~/Temp`
assuming there is `Temp` dir under the home directory

**Comment by FrankChen021** [comment]
That makes senses. Could you also update your findings in the PR?

**Comment by vimil-saju** [comment]
> That makes senses. Could you also update your findings in the PR?

I have updated the PR with findings as well as your proposed changes

**Comment by FrankChen021** [comment]
Closed as #12877 merged
